src/commands/dev/dev.mjs

```diff
@@ -713,7 +822,6 @@ export const createDevCommand = (program) => {
       '--context <context>',
       'Specify a deploy context or branch for environment variables (contexts: "production", "deploy-preview", "branch-deploy", "dev")',
       normalizeContext,
-      'dev',
```

Comment:
By setting a default value here, we don't give commands the opportunity to set their own custom values. I've removed the default and added it in https://github.com/netlify/cli/pull/5338/files#diff-67082469659283bc342bd13f7e20570cb3d1bf5096aad88545a9ac4809d6083bR520.